docs/usage.md

```diff
@@ -0,0 +1,85 @@
+Arduino MKR WiFi 1010, run:
+
+```
+./arduino-fwuploader firmware flash -b arduino:samd:mkrwifi1010 -a /dev/ttyACM0
+```
+
+You just have to specify the fqbn (`-b` or `--fqbn`) of the board and the serial port (`-a` or `--address`) The firmware
+uploader will take care of fetching everything is required to perform the update process. If no module and version are
+specified **the latest version of the firmware** will be used.
+
+If you want to flash a specific version of a firmware you can use the `-m` or `--module` flag
+
+For example to flash the WINC module present on the MKR 1000 with 19.6.1 version of the firmware you can run something
+like:
+
+```
+./arduino-fwuploader firmware flash -b arduino:samd:mkr1000 -a /dev/ttyACM0 -m WINC1500@19.6.1
+```
+
+There is also a retry mechanism bundled in the tool because the flashing process uses serial comunication and sometimes
+can be a bit unreliable. The retry flag is set by default to 9 retries, but it's possible to overwrite it for whatever
+reason. For example to update a Nano RP2040 Connect with the retry set to 2 you can use:
+
+```
+./arduino-fwuploader firmware flash --fqbn arduino:mbed_nano:nanorp2040connect -a /dev/ttyACM0 --retries 2
+```
+
+It's possible also to list the available firmwares for every board/module with:
+
+```
+./arduino-fwuploader firmware list
+```
+
+but you can also filter the results by specifying the `-b` or `--fqbn` flag
+
+The tool offers the ability to print output in json, with the `--format json`
+
+### Certificates
+
+The tool offers also the ability to flash SSL certificates to a module:
+
+```
+./arduino-fwuploader certificates flash -b arduino:samd:nano_33_iot" -a COM10 -u arduino.cc:443 -u google.cc:443
+```
+
+or you can specify a path to a file with `-f` instead of the url of the certificate
+
+### Command line options
+
+The full list of command line options can be obtained with the `-h` option: `./arduino-fwuploader -h`
+
+```
+Arduino Firmware Uploader (arduino-fwuploader).
+
+Usage:
+  arduino-fwuploader [command]
+
+Examples:
+  ./arduino-fwuploader <command> [flags...]
+
+Available Commands:
+  certificates Commands to operate on certificates.
+  firmware     Commands to operate on firmwares.
+  help         Help about any command
+  version      Shows version number of arduino-fwuploader.
+
+Flags:
+      --format string       The output format, can be {text|json}. (default "text")
+  -h, --help                help for arduino-fwuploader
+      --log-file string     Path to the file where logs will be written
+      --log-format string   The output format for the logs, can be {text|json}.
+      --log-level string    Messages with this level and above will be logged. Valid levels are: trace, debug, info, warn, error, fatal, panic (default "info")
+  -v, --verbose             Print the logs on the standard output.
+
+Use "arduino-fwuploader [command] --help" for more information about a command.
+```
+
```

Comment:
```suggestion
```
I think this is too difficult to maintain. Even though it is only intended to demonstrate the use of the `-h` flag, the readers may end up using it as a reference, but it will likely become outdated over time (as is the case with the equivalent content in Arduino CLI's docs).